apps/builder/app/builder/features/props-panel/controls/url.tsx

```diff
@@ -211,9 +222,94 @@ const BaseAttachment = ({ prop, onChange, onSoftDelete }: BaseControlProps) => (
   </Row>
 );
 
+// store that contains IDs off all instances in the selected page
+const pageInstancesStore = computed(
+  [selectedPageStore, instancesStore],
+  (page, instances) => {
+    const result = new Set<string>();
+
+    if (page === undefined) {
+      return result;
+    }
+
+    const addInstance = (id: string) => {
+      const instance = instances.get(id);
+      if (instance === undefined) {
+        return;
+      }
+      result.add(id);
+      for (const child of instance.children) {
+        if (child.type === "id") {
+          addInstance(child.value);
+        }
+      }
+    };
+    addInstance(page.rootInstanceId);
+
+    return result;
+  }
+);
+
+let sectionsStore = computed(
+  [pageInstancesStore, propsStore],
+  (pageInstances, props) => {
+    const sections = new Map<Instance["id"], string>();
+
+    for (const prop of props.values()) {
+      if (
+        prop.type === "string" &&
+        prop.name === "id" &&
+        prop.value.trim() !== "" &&
+        pageInstances.has(prop.instanceId)
+      ) {
+        sections.set(prop.instanceId, prop.value);
+      }
+    }
+
+    return sections;
+  }
+);
+
+// too hard to mock all the stores that sectionsStore is derived from
+// so we have this for Storybook
+export const setMockUrlSectionsStore = (store: typeof sectionsStore) => {
```

Comment:
another option is to pass the sections to a component <Components sections={sectionsStore.get()} /> 
So this way when you need to mock it, you pass it directly to the component, I just ended up doing this in BreakpointsSelector

I created a BreakpointsSelectorContainer that holds the stores, so I can have a pure BreakpointsSelector I can render in stories

Reply:
In this case we would have to pass this prop to `<PropsPanel>`, which feels front from the separations of concerns point of view.

Reply:
yeah, not great, but we should have a general internal standard for mocking stores, so that we don't end up having hundreds of mock functions

Reply:
maybe @TrySound has an opinion on this